common/service/config/metrics_test.go

```diff
@@ -61,12 +64,24 @@ func (s *MetricsSuite) TestM3() {
 	}
 	config := new(Metrics)
 	config.M3 = m3
-	scope := config.NewScope()
+	scope := config.NewScope(loggerimpl.NewNopLogger())
+	s.NotNil(scope)
+}
+
+func (s *MetricsSuite) TestPrometheus() {
+	prom := &prometheus.Configuration{
+		OnError:       "panic",
+		TimerType:     "histogram",
+		ListenAddress: "127.0.0.1:8002",
```

Comment:
avoid using fixed port numbers in unit test - breaks on a shared build machine with bind_address error. can you use 127.0.0.1:0 for this one ?

Reply:
done